lib/core.js

```diff
@@ -247,6 +247,27 @@ module.exports = class Core {
     }
   }
 
+  async clear (start, end) {
+    await this._mutex.lock()
+
+    this.bitfield.setRange(start, end - start, false)
+
+    start = this.bitfield.lastSet(start) + 1
+    end = this.bitfield.firstSet(end)
+
+    if (end === -1) end = this.tree.length
+
+    try {
+      const offset = await this.tree.byteOffset(start * 2)
+      const [byteEnd, byteEndLength] = await this.tree.byteRange((end - 1) * 2)
+      const length = (byteEnd + byteEndLength) - offset
+
+      await this.blocks.clear(offset, length)
```

Comment:
call the onupdate hook from here. the bitfield option has enough info to encapsulate the drop here. then the timing of that is locked on the mutex also